surfactant/infoextractors/js_file.py

```diff
@@ -27,56 +29,38 @@
 DATABASE_URL = "https://raw.githubusercontent.com/RetireJS/retire.js/master/repository/jsrepository-master.json"
 
 
-class JSDatabaseManager:
+class JSDatabaseManager(BaseDatabaseManager):
+    """Manages the JavaScript library database."""
+
     def __init__(self):
-        self._js_lib_database: Optional[Dict[str, Any]] = None  # Use the private attribute
-        self.database_version_file_path = (
-            ConfigManager().get_data_dir_path()
-            / "infoextractors"
-            / "js_library_patterns"
-            / "js_library_patterns.toml"
+        super().__init__(
+            pattern_key="js_library_patterns",
+            pattern_file="js_library_patterns.json",
+            source="jsfile.retirejs",
         )
-        self.pattern_key = "js_library_patterns"
-        self.pattern_file = "js_library_patterns.json"
-        self.source = "jsfile.retirejs"
-        self.new_hash: Optional[str] = None
-        self.download_timestamp: Optional[datetime] = None
-
-    @property
-    def js_lib_database(self) -> Optional[Dict[str, Any]]:
-        if self._js_lib_database is None:
-            self.load_db()
-        return self._js_lib_database
 
     @property
-    def pattern_info(self) -> Dict[str, Any]:
-        return {
-            "pattern_key": self.pattern_key,
-            "pattern_file": self.pattern_file,
-            "source": self.source,
-            "hash_value": self.new_hash,
-            "timestamp": self.download_timestamp,
-        }
-
-    def load_db(self) -> None:
-        js_lib_file = (
-            ConfigManager().get_data_dir_path()
-            / "infoextractors"
-            / "js_library_patterns"
-            / self.pattern_file
-        )
-
-        try:
-            with open(js_lib_file, "r") as regex:
-                self._js_lib_database = json.load(regex)
-        except FileNotFoundError:
-            logger.warning(
-                "Javascript library pattern database could not be loaded. Run `surfactant plugin update-db js_file` to fetch the pattern database."
-            )
-            self._js_lib_database = None
-
-    def get_database(self) -> Optional[Dict[str, Any]]:
-        return self._js_lib_database
+    def data_dir(self) -> Path:
+        """Returns the base directory for storing JavaScript library database files."""
+        return ConfigManager().get_data_dir_path() / "infoextractors" / "js_library_patterns"
```

Comment:
I think if the parent class handles returning `ConfigManager().get_data_dir_path()`, then this can just call `super() / js_library_patterns`. We can probably also remove the `infoextractors` subfolder so things aren't nested as deeply -- I don't think plugins with databases will be so plentiful that we'll need to worry about running out of names in the top level data_dir folder.

Same for the native lib data_dir function.